File: tomgo/tests/testdata/equiv/list/scenario.go
// Equivalence scenario for List.gom (variadic of built-in int).
package main

import (
	"fmt"

	"tomgen/list"
)

func main() {
	a := list.MakeConc(1, 2, 3)
	b := list.MakeConc(1, 2, 3)
	c := list.MakeConc(1, 2, 4)

	fmt.Printf("a=%v\n", a)
	fmt.Printf("shared=%v\n", a == b)
	fmt.Printf("diff=%v\n", a != c)

	// Empty list — built via MakeConc with no args on the Go side.
	empty := list.MakeConc()
	fmt.Printf("empty=%v\n", empty)
	fmt.Printf("shared-empty=%v\n", empty == list.MakeConc())

	// Length via the slice field length (Go) — we just print it.
	fmt.Printf("len=%d\n", len(a.(*list.ConcList).Slots))
}

File: tomgo/tests/reports/equiv-list/java-src/Scenario.java
// Equivalence scenario for List.gom — Java reference side.
import gom.list.types.List;
import gom.list.types.list.conc;
import gom.list.types.list.Emptyconc;

public class Scenario {
    public static void main(String[] args) {
        List a = conc.fromArray(new int[]{1, 2, 3});
        List b = conc.fromArray(new int[]{1, 2, 3});
        List c = conc.fromArray(new int[]{1, 2, 4});

        System.out.println("a=" + a);
        System.out.println("shared=" + (a == b));
        System.out.println("diff=" + (a != c));

        List empty = Emptyconc.make();
        System.out.println("empty=" + empty);
        System.out.println("shared-empty=" + (empty == Emptyconc.make()));

        // length() walks the cons-list; matches the Go side's slice len.
        System.out.println("len=" + ((conc) a).length());
    }
}
